src/MoBi.Presentation/Mappers/CreateModuleDTOToModuleMapper.cs

```diff
@@ -55,32 +47,39 @@ public class CreateModuleDTOToModuleMapper : ModuleDTOMapper, ICreateModuleDTOTo
          _spatialStructureFactory = spatialStructureFactory;
       }
 
+      private void conditionalAdd<T>(Module module, T buildingBlock) where T : class, IBuildingBlock
+      {
+         if (buildingBlock == null)
+            return;
+         
+         module.Add(buildingBlock);
+      }
+
       public Module MapFrom(CreateModuleDTO createModuleDTO)
       {
          var module = _context.Create<Module>().WithIcon(ApplicationIcons.Module.IconName).WithName(createModuleDTO.Name);
 
-         module.Add(conditionalCreate(createModuleDTO.WithMolecule,
```

Comment:
the conditionalCreate used to assign to the member, but now calls the Add method. We just need to not add null bb's now